pkg/columns/formatter/json/json.go

```diff
@@ -124,39 +130,92 @@ func NewFormatter[T any](cols columns.ColumnMap[T], options ...Option) *Formatte
 	return tf
 }
 
-// FormatEntry returns an entry as a formatted string, respecting the given formatting settings
-func (f *Formatter[T]) FormatEntry(entry *T) string {
-	if entry == nil {
-		return ""
-	}
+func (f *Formatter[T]) printCols(entry *T, buf *encodeState, cols []*column[T], level int) {
+	indent := strings.Repeat("  ", level+1)
+	colsWithParent := map[string][]*column[T]{}
+	colsWithoutParent := []*column[T]{}
 
-	buf := bufpool.Get().(*encodeState)
-	buf.Reset()
-	defer bufpool.Put(buf)
-
-	buf.WriteByte('{')
-	if f.options.prettPrint {
-		buf.WriteByte('\n')
+	for _, col := range cols {
+		// Save columns which has a parent in its name
+		if strings.Count(col.column.Name, ".") > level {
+			hierarchy := strings.Split(col.column.Name, ".")
+			parentName := hierarchy[level]
+			colsWithParent[parentName] = append(colsWithParent[parentName], col)
+			continue
+		}
+		// Don't print columns without a parent directly
+		// It is possible that their name is the same as a name of an object in the same level
+		// For example: {"a": {"bbb": 1}, "a": 2}
```

Comment:
I don't get this point. Wouldn't it be an error to have two things under the same key? (Even if one of them is an object and the other is a column).

Reply:
For JSON it is an error. For the columns library it is not an error.

See the KubeIPResolver PR: It has a virtual column called `src` which displays something like `p/test-pod:8080`. But the Endpoint struct is also called `src` and its therefore its children are called `src.port: 8080`, etc...

Reply:
You're right. Then the current logic makes sense.